### DISCUSSION TITLE
Issue with convert value string into std::string and vice versa

### DISCUSSION THREAD

**Discussion by tuongbuimtt** [yosh-matsuda/cpp-yyjson#16] [Q&A] [answered]
Sometimes my output JSON has an invalid utf-8 string.
It seems to appear in a random position. My JSON has many nested fields, and adding one by one. 
I mainly write values from std::string.
Do you have any advice for my case?
```
"sid":"1420\u0000\u0000\u0000\u00003649932"
the correct string is 14203649932
```
Thanks.

**Comment by tuongbuimtt** [comment]
Update: 
I found myself that when I use functions like
```
string sid = "123456";
json.emplace("sid", sid);
```
will lead to my problem.

So I fixed that:
```
string sid = "123456";
json.emplace("sid", value(sid, copy_string));
```
And it works.
So I would like to suggest a new interface `emplace` that supports adding std::string directly for coding convenience.

**Reply by yosh-matsuda** [reply, marked as answer]
The `emplace` member function has the `copy_string`  specifier in the 3rd argument.   
Does this also solve your problem?

```cpp
json.emplace("sid", sid, copy_string);
```

**Reply by tuongbuimtt** [reply]
I have tried
```
json.emplace("sid", sid, copy_string);
```
It works fine. :+1:  thank for your support

**Comment by tuongbuimtt** [comment]
I have a question-answer When I tried:
```
std::string key = "the key";
json.emplace(key, 123);
```
Cause the segment faults error.
Is there a way to add a key from std::string? Because sometimes, the key can be modified many times, and I cannot always assign a const char* type for  the key.

Update:
```
std::string key = "the key";
json.emplace(key, 123, copy_string);
```
get helpful. :+1:
